boards: arm: mimxrt1170_evkb: add RT1170 EVKB

danieldegrasse opened the pull request:
Add RT1170 EVKB definition to Zephyr. This board is similar to the
RT1170 EVK, but uses a different quadspi flash chip, and includes
an audio connector as well as an additional display connection.

The following features have been tested with this port:
The following features have been verified to function on this board:
- GPIO 
- COUNTER
- CAN
- SPI
- I2C
- PWM
- ADC
- UART 
- DMA
- GPT
- WATCHDOG
- USB
- HWINFO
- DISPLAY
- CAAM
- SDHC
- Multicore boot

The following features are not currently tested/enabled:
- ENET (This board uses an unsupported PHY)
- SAI (Should work, but requires board modification to verify)
- ACMP (Untested)
- FLEXSPI (This board's flash chip is not supported)


Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: arm: mimxrt1170_evkb: add RT1170 EVKB

Add RT1170 EVKB definition to Zephyr. This board is similar to the
RT1170 EVK, but uses a different quadspi flash chip, and includes
an audio connector as well as an additional display connection.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: arm: mixmrt1170_evkb: add support for ADC

Add support for ADC sample and API test on RT1170 EVKB.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: arm: mimxrt1170_evkb: add support for DMA test

Enable DMA test for RT1170 EVKB

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: arm : mimxrt1170_evkb: add support for I2C

Add support for I2C on RT1170 EVKB, verified using I2C target API test.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: arm: mimxrt1170_evkb: add support for SPI

Add support for SPI on RT1170 EVKB, verified using SPI loopback sample.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: arm: mixmrt1170_evkb: enable display support

Enable display support for RT1170 EVKB, using display sample with
RK055HDMIPI4M display.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
samples: ipc: enable openamp sample for RT1170 EVKB

Enable openamp sample for RT1170 EVKB.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

DerekSnell commented on samples/subsys/ipc/openamp/boards/mimxrt1170_evkb_cm7.conf:
```diff
@@ -0,0 +1,2 @@
+CONFIG_INCLUDE_REMOTE_DIR=y
```
```suggestion
#
# Copyright 2023, NXP
#
# SPDX-License-Identifier: Apache-2.0
#

CONFIG_INCLUDE_REMOTE_DIR=y
```

DerekSnell commented on samples/subsys/ipc/openamp/remote/boards/mimxrt1170_evkb_cm4.conf:
```diff
@@ -0,0 +1,3 @@
+CONFIG_BUILD_OUTPUT_INFO_HEADER=y
```
```suggestion
#
# Copyright 2023, NXP
#
# SPDX-License-Identifier: Apache-2.0
#

CONFIG_BUILD_OUTPUT_INFO_HEADER=y
```

DerekSnell reviewed (changes requested):
Hi @danieldegrasse ,
Thank you for contributing this board.  Overall, this LGTM.  However, we also need to update the board doc page, and clarify both boards are supported.  Since these are in the same board folder and same doc page, we should be clear which features are supported on both boards.  Perhaps the Supported Features table should have separate columns: one for the new `EVKB`  revision, and another the original `EVK` revision.

Thanks

danieldegrasse commented:
> 

Hi @danieldegrasse ,
Thank you for contributing this board. Overall, this LGTM. However, we also need to update the board doc page, and clarify both boards are supported. Since these are in the same board folder and same doc page, we should be clear which features are supported on both boards. Perhaps the Supported Features table should have separate columns: one for the new EVKB revision, and another the original EVK revision.

Thanks

Thanks for highlighting this @DerekSnell. I have added a table as you suggested, and tracked the status of features on each board. I have also added a "(M7)" specifier for any features that are only validated on the M7 core.

DerekSnell reviewed (approved):
LGTM, @danieldegrasse .  Thanks for updating the doc page

mmahadevan108 commented on samples/drivers/display/boards/mimxrt1170_evkb_cm7.conf:
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_HEAP_MEM_POOL_SIZE=4194304
```
WHy is this changed needed for EVKB and not EVK?
